deeplinkdispatch/src/main/java/com/airbnb/deeplinkdispatch/BaseDeepLinkDelegate.kt

```diff
@@ -0,0 +1,442 @@
+        }
+        val handlerArgsConstructorParams = createParamArray(typeNameMap, result.parameters)
+        val handlerParameters =
+            handlerParameterClazzConstructor.newInstance(*handlerArgsConstructorParams)
+        val handlerInstanceField = handlerClazz.getField("INSTANCE").get(null);
+        handlerMethod.invoke(handlerInstanceField, handlerParameters)
+    }
+
+    private fun createParamArray(
+        typeNameMap: List<Pair<DeeplinkParam, Class<*>>>,
+        parameters: Map<String, String>
+    ): Array<Any?> {
+        return typeNameMap.map { (annotation, type) ->
+            when(annotation.type) {
+                DeepLinkParamType.Path ->
+                    mapNotNullableType(parameters.getOrElse(annotation.name) {error("Non existent non nullable element for name: ${annotation.name}")} , type)
+                DeepLinkParamType.Query ->
+                    mapNullableType(parameters[annotation.name], type)
+            }
+        }.toTypedArray()
+    }
+
+    private fun mapNullableType(value: String?, type: Class<*>): Any? {
+        if (value == null) return null
+        return try {
+            when (type) {
+                Boolean::class.javaObjectType -> value.toBoolean()
+                Int::class.javaObjectType -> value.toInt()
+                Long::class.javaObjectType -> value.toLong()
+                Short::class.javaObjectType -> value.toShort()
+                Byte::class.javaObjectType -> value.toByte()
+                Double::class.javaObjectType -> value.toDouble()
+                Float::class.javaObjectType -> value.toFloat()
+                else -> value
+            }
+        } catch (e: NumberFormatException) {
+            null
+        }
+    }
+
+    private fun mapNotNullableType(value: String, type: Class<*>): Any {
+        return try {
+            when (type) {
+                Boolean::class.javaPrimitiveType -> value.toBoolean()
+                Int::class.javaPrimitiveType -> value.toInt()
+                Long::class.javaPrimitiveType -> value.toLong()
+                Short::class.javaPrimitiveType -> value.toShort()
+                Byte::class.javaPrimitiveType -> value.toByte()
+                Double::class.javaPrimitiveType -> value.toDouble()
+                Float::class.javaPrimitiveType -> value.toFloat()
+                else -> value
+            }
+        } catch (e: NumberFormatException) {
+            0
+        }
+    }
+
+    /**
+     * Create a [DeepLinkResult], whether we are able to match the uri on
+     * {@param sourceIntent} or not.
+     *
+     * @param activity      used to startActivity() or notifyListener().
+     * @param sourceIntent  inbound Intent.
+     * @param deeplinkMatchResult deeplinkMatchResult that matches the sourceIntent's URI. Derived from
+     * [.findEntry]. Can be injected for testing.
+     * @return DeepLinkResult
+     */
+    fun createResult(
```

Comment:
Do we need to make dispatchResult public? Or an extension of `DeepLinkResult`? I'm guessing a use case may be to call `createResult` and the conditionally dispatch, and it would be nice to avoid going through the parsing logic a second time

Reply:
I'll do that as a separate PR as discussed offline.